tests/gtest/avifrgbtoyuvtest.cc

```diff
@@ -0,0 +1,350 @@
+            const int64_t diff = static_cast<int64_t>(rowB[x]) - static_cast<int64_t>(rowA[x]);
+            *diffSum += diff;
+            *absDiffSum += std::abs(diff);
+            *sqDiffSum += diff * diff;
+            *maxAbsDiff = std::max(*maxAbsDiff, std::abs(diff));
+        }
+    }
+}
+
+void getDiffSumAndSqDiffSum(const avifRGBImage & a, const avifRGBImage & b, int64_t * diffSum, int64_t * absDiffSum, int64_t * sqDiffSum, int64_t * maxAbsDiff)
+{
+    (a.depth <= 8) ? getDiffSumAndSqDiffSum<uint8_t>(a, b, diffSum, absDiffSum, sqDiffSum, maxAbsDiff)
+                   : getDiffSumAndSqDiffSum<uint16_t>(a, b, diffSum, absDiffSum, sqDiffSum, maxAbsDiff);
+}
+
+// Returns the Peak Signal-to-Noise Ratio from accumulated stats.
+double getPsnr(double sqDiffSum, double numDiffs, double maxAbsDiff)
+{
+    if (sqDiffSum == 0.)
+        return 99.; // Lossless.
+    const double distortion = sqDiffSum / (numDiffs * maxAbsDiff * maxAbsDiff);
+    return (distortion > 0.) ? std::min(-10 * std::log10(distortion), 98.9) : 98.9; // Not lossless.
+}
+
+//------------------------------------------------------------------------------
+
+class YUVToRGBTest
+    : public testing::TestWithParam<
+          std::tuple</*rgbDepth=*/int, /*yuvDepth=*/int, avifRGBFormat, avifPixelFormat, avifRange, avifMatrixCoefficients, /*plainColor=*/bool, /*rgbStep=*/uint32_t, /*maxAbsAverageDiff=*/double, /*minPsnr=*/double>>
+{
+};
+
+// Converts from RGB to YUV and back to RGB for all RGB combinations, separated by a color step for reasonable timing.
+// Unless plainColor is true, also applies some noise to the input samples to exercise chroma subsampling.
+TEST_P(YUVToRGBTest, Convert)
+{
+    const int rgbDepth = std::get<0>(GetParam());
+    const int yuvDepth = std::get<1>(GetParam());
+    const avifRGBFormat rgbFormat = std::get<2>(GetParam());
+    const avifPixelFormat yuvFormat = std::get<3>(GetParam());
+    const avifRange yuvRange = std::get<4>(GetParam());
+    const avifMatrixCoefficients matrixCoefficients = std::get<5>(GetParam());
+    // Whether to add noise to the input RGB samples. Should only impact subsampled chroma (4:2:2 and 4:2:0).
+    const bool plainColor = std::get<6>(GetParam());
+    // Testing each RGB combination would be more accurate but results are similar with faster settings.
+    const uint32_t rgbStep = std::get<7>(GetParam());
+    // Thesholds to pass.
+    const double maxAbsAverageDiff = std::get<8>(GetParam());
+    const double minPsnr = std::get<9>(GetParam());
+    // Deduced constants.
+    const bool isMonochrome = (yuvFormat == AVIF_PIXEL_FORMAT_YUV400); // If true, only test greyish input.
+    const uint32_t rgbMax = (1 << rgbDepth) - 1;
+
+    // The YUV upsampling treats the first and last rows and columns differently than the remaining pairs of rows and columns.
+    // An image of 16 pixels is used to test all these possibilities.
+    const int width = 4;
+    const int height = 4;
```

Comment:
these could be constexpr int kWidth (assuming that naming convention is common)

Reply:
Naming convention for constants in libavif seems to be [lowercase](https://github.com/AOMediaCodec/libavif/blob/e6368422d891ca940ed9f77a6c5e80d26191f280/src/write.c#L33). It may change to kSomething in C++ files when the Google C++ style guide is applied to them.